src/main/java/deadline.java

```diff
@@ -0,0 +1,24 @@
+public class deadline extends task {
+
+    protected String date;
+
+    public deadline(String name, Integer number, boolean isDone, String taskType) {
+        super(name, number, isDone, taskType);
+        this.date = name.substring(name.indexOf('/')+1);
+    }
+
+    @Override
+    public String date() {
+        return name.substring(name.indexOf('/'));
+    }
+
+    @Override
+    public String getName() {
+        return name.substring(9, name.indexOf('/'));
```

Comment:
Try to avoid magic numbers.
Others may not know why is 9 used
Consider rethinking the logic of the program